vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -348,9 +348,13 @@ def collect_metrics_async(self):
             self.log.debug("Queued all %d tasks, waiting for completion.", len(tasks))
             for future in as_completed(tasks):
                 e = future.exception()
-                if e is not None:
+                if isinstance(e, vmodl.fault.InvalidArgument):
```

Comment:
It looks like this exception is thrown in a lot of different cases. Do you think it's likely we'll ever get this exception in cases where we aren't querying metrics that don't exist?

Reply:
Unfortunately I can't make it more specific. Other cases would be that the query we make is invalid.